File: packages/reflex-shared-types/src/tracking-server-app-settings.ts
import { FilterSettings } from './config/filter-settings';
import { Calibration } from './config/calibration';
import { CameraConfiguration } from './config/camera-configuration';
import { NetworkSettings } from './config/network-settings';
import { ProcessingSettings } from './config/processing-settings';
import { TuioConfiguration } from './config/tuio-configuration';
import { RemoteProcessingServiceSettings } from './config/remote-processing-service-settings';
import { PointCloudSettings } from './config/point-cloud-settings';

export interface TrackingServerAppSettings {

  filterSettingValues: FilterSettings;
  calibrationValues: Calibration;
  cameraConfigurationValues: CameraConfiguration;
  networkSettingValues: NetworkSettings;
  pointCloudSettingValues: PointCloudSettings;
  processingSettingValues: ProcessingSettings;
  remoteProcessingServiceSettingsValues: RemoteProcessingServiceSettings;
  tuioSettingValues: TuioConfiguration;
  isAutoStartEnabled: boolean;
  defaultCamera: string;
}

File: tools/ReFlex.TrackingServer/Data/TrackingServerAppSettings.cs
using System;
using System.Diagnostics.CodeAnalysis;
using ReFlex.Core.Tuio.Util;
using TrackingServer.Data.Config;

namespace TrackingServer.Data
{
    public class TrackingServerAppSettings
    {
        public FilterSettings FilterSettingValues { get; set; } = new();

        public ReFlex.Core.Calibration.Util.Calibration CalibrationValues { get; set; }

        public FrameSizeDefinition FrameSize { get; set; } = new();

        public bool IsAutoStartEnabled { get; set; }

        public string DefaultCamera { get; set; } = "";

        public CameraConfiguration CameraConfigurationValues { get; set; } = new();

        public NetworkSettings NetworkSettingValues { get; set; } = new();

        public PredictionSettings PredictionSettings { get; set; } = new();

        public ProcessingSettings ProcessingSettingValues { get; set; } = new();

        public RemoteProcessingServiceSettings RemoteProcessingServiceSettingsValues { get; set; } = new();

        public TuioConfiguration TuioSettingValues { get; set; } = new();

        public string GetCompleteValues()
        {
            var result = $"{DefaultCamera} | ";
            result += $"{CameraConfigurationValues.GetCameraDefinitionString()}";
            result += $"{nameof(IsAutoStartEnabled)}: {IsAutoStartEnabled}{Environment.NewLine}";
            result += FrameSize?.GetFrameSizeDefinitionString() ?? $"{nameof(FrameSize)} not set.";
            result += FilterSettingValues?.GetFilterSettingsString() ?? $"{nameof(FilterSettingValues)} not set.";
            result += CalibrationValues.GetCalibrationValuesString();
            result += NetworkSettingValues?.GetNetworksSettingsString() ?? $"{nameof(NetworkSettingValues)} not set.";
            result += ProcessingSettingValues?.GetPorcessingSettingsString()  ?? $"{nameof(ProcessingSettingValues)} not set.";
            result += RemoteProcessingServiceSettingsValues?.GetRemoteProcessingServiceSettingsString()  ?? $"{nameof(RemoteProcessingServiceSettingsValues)} not set.";
            result += TuioSettingValues?.GetTuioConfigurationString()  ?? $"{nameof(TuioSettingValues)} not set.";

            return result;
        }
    }
}